tests/common/test_additional_secrets_redaction.py

```diff
@@ -0,0 +1,402 @@
+    # ============================================================================
+    @patch("pyracf.common.irrsmo00.IRRSMO00.call_racf")
+    def test_resource_admin_custom_secret_redacted_on_success(
+        self,
+        call_racf_mock: Mock,
+    ):
+        resource_admin = ResourceAdmin(additional_secret_traits=["base:owner"])
+        call_racf_mock.side_effect = [
+            TestResourceConstants.TEST_EXTRACT_RESOURCE_RESULT_BASE_SUCCESS_XML,
+            TestResourceConstants.TEST_ALTER_RESOURCE_RESULT_SUCCESS_XML,
+        ]
+        result = resource_admin.alter(
+            "TESTING",
+            "ELIJTEST",
+            traits=TestResourceConstants.TEST_ALTER_RESOURCE_REQUEST_TRAITS,
+        )
+        self.assertEqual(
+            result,
+            TestCommonConstants.TEST_ALTER_RESOURCE_RESULT_SUCCESS_OWNER_SECRET_DICTIONARY,
+        )
+        self.assertNotIn(
+            TestResourceConstants.TEST_ALTER_RESOURCE_REQUEST_TRAITS["base:owner"],
+            str(result),
+        )
+
+    @patch("pyracf.common.irrsmo00.IRRSMO00.call_racf")
+    def test_resource_admin_custom_mapped_secret_redacted_on_success(
+        self,
+        call_racf_mock: Mock,
+    ):
+        resource_admin = ResourceAdmin(additional_secret_traits=["base:audit_update"])
+        call_racf_mock.side_effect = [
+            TestResourceConstants.TEST_EXTRACT_RESOURCE_RESULT_BASE_SUCCESS_XML,
+            TestCommonConstants.TEST_ALTER_RESOURCE_OVERWRITE_AUDIT_RESULT_SUCCESS_XML,
+        ]
+        result = resource_admin.overwrite_audit_rules_by_access_level(
+            "TESTING",
+            "ELIJTEST",
+            update="ALL",
+        )
+        self.assertEqual(
+            result,
+            TestCommonConstants.TEST_ALTER_RESOURCE_RESULT_SUCCESS_AUDIT_SECRET_DICTIONARY,
+        )
+        self.assertNotIn(
+            "ALL",
+            str(result),
+        )
+
+    @patch("pyracf.common.irrsmo00.IRRSMO00.call_racf")
+    def test_resource_admin_custom_secret_redacted_on_error(
+        self,
+        call_racf_mock: Mock,
+    ):
+        secret_trait = "base:universal_access"
+        resource_admin = ResourceAdmin(additional_secret_traits=[secret_trait])
+        call_racf_mock.side_effect = [
+            TestResourceConstants.TEST_EXTRACT_RESOURCE_RESULT_BASE_SUCCESS_XML,
+            TestResourceConstants.TEST_ALTER_RESOURCE_RESULT_ERROR_XML,
+        ]
+        with self.assertRaises(SecurityRequestError) as exception:
+            resource_admin.alter(
+                "TESTING",
+                "ELIJTEST",
+                traits=TestResourceConstants.TEST_ALTER_RESOURCE_REQUEST_ERROR_TRAITS,
+            )
+        self.assertEqual(
+            exception.exception.result,
+            TestCommonConstants.TEST_ALTER_RESOURCE_RESULT_ERROR_UACC_SECRET_DICTIONARY,
+        )
+        self.assertNotIn(
+            TestResourceConstants.TEST_ALTER_RESOURCE_REQUEST_ERROR_TRAITS[
+                f"{secret_trait}"
```

Comment:
Change `f"{secret_trait}"` to `secret_trait`.

Reply:
Again, not sure why I did that at all, but changed.